tests/Base.php

```diff
@@ -207,14 +212,40 @@ public function testRequestMethod(): void
 
         $response = $this->execute(method: 'OPTIONS', headers: ['x-action' => 'requestMethod']);
         self::assertEquals(200, $response['code']);
-        // Bug in C++ framework makes this an empty string
-        // self::assertEquals('OPTIONS', $response['body']);
+
+        if($this->runtime === 'cpp') {
+            // Bug in C++ framework
+            self::assertEquals('', $response['body']);
+        } else {
+            self::assertEquals('OPTIONS', $response['body']);
+        }
 
         $response = $this->execute(method: 'PATCH', headers: ['x-action' => 'requestMethod']);
         self::assertEquals(200, $response['code']);
         self::assertEquals('PATCH', $response['body']);
     }
 
+    public function testHeadAndTraceRequestMethods(): void
+    {
+        $response = $this->execute(method: 'HEAD', headers: ['x-action' => 'requestHeadOrTrace']);
+        self::assertEquals(200, $response['code']);
+
+        // head, trace dont work well with cpp [drogon].
+        if($this->runtime === 'cpp') return;
+
+        self::assertStringContainsString('HEAD', $response['headers']['x-open-runtimes-logs']);
+
+        $response = $this->execute(method: 'TRACE', headers: ['x-action' => 'requestHeadOrTrace']);
+        if($this->runtime === 'php') {
+            // Bug in PHP Swoole
+            self::assertEquals(400, $response['code']);
+        } else {
+            self::assertEquals(200, $response['code']);
+        }
+
+        self::assertStringContainsString('TRACE', $response['headers']['x-open-runtimes-logs']);
```

Comment:
Based on: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/TRACE

Seems like TRACE should be allowed to have body. What's the reason to use logs here?

Reply:
If I remember correctly, we discussed using the logs for both `Head` & `TRACE` as one of the runtimes did not return a body for `TRACE` as well. I will find the exact convo. and ping it again to you.

Reply:
Thank you so much for taking time to ensure quality of this repository 🙏
Let's switch back to body response, and have `if` to skip the test for the runtime that doesn't support it.

Later we can fix those separately, but using proper body test ensures quality for the rest of runtimes.

Reply:
Sure thing.

Reply:
As discussed, most likely the web servers wouldn't really return anything. Hence the response body check is not added.